sdk/storage/AzureStorageBlob/Source/StorageStreamDownloader.swift

```diff
@@ -160,12 +160,16 @@ internal class ChunkDownloader {
         let modifiedAccessConditions = options.modifiedAccessConditions
         let cpk = options.customerProvidedEncryptionKey
 
-        headers[.xmsRange] = "bytes=\(startRange)-\(endRange)"
         if let rangeGetContentMD5 = options.range?.calculateMD5 {
             headers[.rangeGetContentMD5] = String(rangeGetContentMD5)
+            headers[.range] = "bytes=\(startRange)-\(endRange)"
         }
         if let rangeGetContentCRC64 = options.range?.calculateCRC64 {
             headers[.rangeGetContentCRC64] = String(rangeGetContentCRC64)
+            headers[.range] = "bytes=\(startRange)-\(endRange)"
+        }
+        if headers[.range] == nil {
+            headers[.xmsRange] = "bytes=\(startRange)-\(endRange)"
```

Comment:
To use the hashing functions you must use `.range`, not `.xmsRange` or you'll get a 400 error.

Reply:
Is it safe to just use `range` all the time? The API docs state:
> If the x-ms-range header is specified on a request, then the service uses the range specified by x-ms-range; otherwise, the range specified by the Range header is used.

Reply:
Since you changed it to x-ms-range in your PR, I assumed there was some case for it so I kept that. But when requesting hashes, you _must_ use Range.

Reply:
Oh, no I just realized that we had `HTTPHeaders.range = "Range"` and `StorageHTTPHeaders.range = "x-ms-range"`, I simply renamed the enum value for the second to `.xmsRange` so that places where it was used didn't need to qualify the value.

Reply:
I didn't see any issues when I set both headers to the same value, so I went with that. Let me know what you think.

Reply:
It's a small thing so I don't think it matters, but I guess for simplicity's sake if we can just use Range, I'd prefer that. Probably just so that coming back to this code later, someone (maybe me :p) doesn't look at it and ask "okay why are we setting both headers to the same value??".

Reply:
Updated.